test/cql-pytest/test_type_date.py

```diff
@@ -56,17 +56,20 @@ def test_type_date_from_int_prepared(cql, table1):
 # one that doesn't fit 32 bits, is an error. Cassandra says:
 # "Unable to make unsigned int (for date) from: '-123'".
 # Reproduces #17066.
-@pytest.mark.xfail(reason="#17066")
 def test_type_date_from_int_unprepared_underflow(cql, table1):
     p = unique_key_int()
-    with pytest.raises(InvalidRequest, match="unsigned int"):
+    with pytest.raises(InvalidRequest, match="overflows"):
         cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, -123)")
 
-@pytest.mark.xfail(reason="#17066")
 def test_type_date_from_int_unprepared_overflow(cql, table1):
     p = unique_key_int()
-    with pytest.raises(InvalidRequest, match="unsigned int"):
-        cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, 50000000000000000)")
+    with pytest.raises(InvalidRequest, match="overflows"):
+        cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, 50000000000)")
+
+def test_type_date_from_int_unprepared_out_of_range(cql, table1):
+    p = unique_key_int()
+    with pytest.raises(InvalidRequest, match="large"):
+        cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, 50000000000000000000)")
```

Comment:
It will not be clear to readers what's the difference between these two tests for "overflow" and "out_of_range". I would make this just one test, with a look, e.g.,
```python
for val in ["50000000000", "50000000000000000000"]:
    with pytest.raises(InvalidRequest, match="unsigned int"):
        cql.execute(f"INSERT INTO {table1} (p, d) VALUES ({p}, {val})")
```
By the way, I also think it's a mistake for the two cases to have different error messages - as far as the user is concerned, it's the same error. Both could use the same error text "Unable to make unsigned int (for date) from: '...'" just like in Cassandra.